std-bits/table/src/main/java/org/enso/table/expressions/ExpressionVisitorImpl.java

```diff
@@ -0,0 +1,342 @@
+package org.enso.table.expressions;
+
+import org.antlr.v4.runtime.BaseErrorListener;
+import org.antlr.v4.runtime.CharStreams;
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.antlr.v4.runtime.RecognitionException;
+import org.antlr.v4.runtime.Recognizer;
+
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.PolyglotException;
+import org.graalvm.polyglot.Value;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.LocalDateTime;
+import java.time.ZonedDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Function;
+
+public class ExpressionVisitorImpl extends ExpressionBaseVisitor<Value> {
+  private static class ThrowOnErrorListener extends BaseErrorListener {
+    public static final ThrowOnErrorListener INSTANCE = new ThrowOnErrorListener();
+
+    @Override
+    public void syntaxError(
+        Recognizer<?, ?> recognizer,
+        Object offendingSymbol,
+        int line,
+        int charPositionInLine,
+        String msg,
+        RecognitionException e)
+        throws SyntaxErrorException {
+      throw new SyntaxErrorException(msg, line, charPositionInLine);
+    }
+  }
+
+  public static class SyntaxErrorException extends RuntimeException {
+    private final int line;
+    private final int column;
+
+    public SyntaxErrorException(String message, int line, int column) {
+      super(message);
+      this.line = line;
+      this.column = column;
+    }
+
+    public int getLine() {
+      return line;
+    }
+
+    public int getColumn() {
+      return column;
+    }
+  }
+
+  public static Value evaluate(
+      String expression,
+      Function<String, Value> getColumn,
+      Function<Object, Value> makeConstantColumn,
+      String moduleName,
+      String typeName,
+      String[] variableArgumentFunctions)
+      throws UnsupportedOperationException, IllegalArgumentException {
+    var lexer = new ExpressionLexer(CharStreams.fromString(expression));
+    lexer.removeErrorListeners();
```

Comment:
Out of curiosity: what error listeners may be registered on a just newly created lexer?

Reply:
It sticks a ConsoleErrorListener on by default.